### PR TITLE
Add AI draft file editor

### LINKED ISSUE #15053: Feature parity of AI draft editor with regular file editor

### PR DESCRIPTION
**Pull request by reteps**
# Description

This adds embedded file viewing/editing to the existing AI draft editor route. Users can open the All files tab, select a draft question file, edit it in Ace, and save without leaving the AI editor or losing the chat sidebar.

The editor tabs are now controlled with `react-bootstrap` and persisted in the URL with `nuqs`, so links such as `?file=question.html&tab=all-files` open directly to the selected file.

The file editor component has been rewritten in react. The file loading now uses tRPC.

- [x] I have disclosed the level of AI assistance used (i.e. none, specific portions, majority of implementation). Majority of implementation.

Closes #15053

# Testing

Not manually tested.

### CODE REVIEW

**Comment by reteps** [comment]
```
 Best split: 3 stacked PRs

  1. Shared file editor refactor
     Purpose: make the existing instructor file editor React-hydrated without changing AI draft behavior.

     Include:
      - apps/prairielearn/src/components/AceFileEditor.tsx
      - apps/prairielearn/src/lib/editableFile.ts
      - apps/prairielearn/src/pages/instructorFileEditor/InstructorFileEditorClient.tsx
      - apps/prairielearn/src/pages/instructorFileEditor/instructorFileEditor.html.ts
      - apps/prairielearn/src/pages/instructorFileEditor/instructorFileEditor.ts
      - bundle registration / old asset deletion

     Keep out:
      - AI draft file browser
      - tRPC draft file APIs
      - FileBrowser embedding changes, unless strictly needed
  2. Embeddable file browser + shared file actions
     Purpose: prepare existing file browser machinery for reuse, while preserving current instructor file browser behavior.

     Include:
      - apps/prairielearn/src/components/FileBrowser.tsx changes for createDirectoryBrowserHtml, createFilePreviewHtml, customizable URLs/actions/attributes
      - apps/prairielearn/src/lib/course-file-actions.ts
      - apps/prairielearn/src/pages/instructorFileBrowser/instructorFileBrowser.ts refactor to use shared actions and Zod parsing

     One important cleanup before making this PR: don’t have course-file-actions.ts import draft-specific logic from draft-question-files.ts. Either omit that guard in PR 2 and add it in
     PR 3, or make course-file-actions accept an optional validation callback. Otherwise PR 2 has a weird dependency on the AI draft feature.
  3. AI draft all-files editor
     Purpose: actual product feature.

     Include:
      - apps/prairielearn/src/lib/draft-question-files.ts
      - apps/prairielearn/src/lib/draft-question-files.test.ts
      - apps/prairielearn/src/trpc/course/ai-draft-files.ts
      - tRPC router wiring
      - AI draft editor page/component changes
      - apps/prairielearn/src/ee/pages/instructorAiGenerateDraftEditor/components/SelectedQuestionFileEditor.tsx
      - CSS changes

  If you only want two PRs

  Do this:

  1. File editor/browser infrastructure
     Combine PRs 1 and 2 above. This is still a coherent “make file editing/browser reusable” PR, but it is a bit larger.
  2. AI draft file editing
     Everything AI-specific: draft question file data, tRPC, tabs/query state, selected file editing, save/preview flow.
```

**Comment by reteps** [comment]
Various improvements were made in newer commits: The file browser is now in react so that code can be better shared. We handle file conflicts in the draft editor. We better reuse the `AceEditor` component in the `question.html` / `server.py` tab. Additional e2e tests for the file editor. Graceful handling of various edge cases.

**Comment by reteps** [comment]
This first review requested is for code architecture. I tested the generalized file editor / draft file editor lightly to ensure there were no obvious and major issues.

File: apps/prairielearn/src/lib/draft-question-files/mutations.ts [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+}: DraftQuestionFileMutationContext & {
+  files: Record<string, string | null>;
+}): Promise<void> {
+  await runDraftEditorJob(
+    new QuestionModifyEditor({
+      locals: { ...editorLocals({ course, user, authn_user, authz_data }), question },
+      files,
+    }),
+  );
+}
```
[the remark this answers is not in the document]
Comment (reteps) [reply apps/prairielearn/src/lib/draft-question-files/mutations.ts]: This is a known issue with the split-view pane. I'll add a TODO for this.

File: apps/prairielearn/src/ee/pages/instructorAiGenerateDraftEditor/components/DraftQuestionFileBrowserActions.tsx [hunk 2]
```diff
@@ -0,0 +1,452 @@
...
+}) {
+  const [show, setShow] = useState(false);
+  const [wasDisabled, setWasDisabled] = useState(disabled);
+
+  // Close the popover if the action becomes disabled while it's open (e.g. an
+  // AI generation starts)
+  if (disabled !== wasDisabled) {
+    setWasDisabled(disabled);
+    if (disabled) setShow(false);
+  }
```
[the remark this answers is not in the document]
Comment (reteps) [reply apps/prairielearn/src/ee/pages/instructorAiGenerateDraftEditor/components/DraftQuestionFileBrowserActions.tsx]: Keeping this as-is. This is React's documented ["adjusting state when a prop changes"](https://react.dev/reference/react/useState#storing-information-from-previous-renders) pattern — a conditional render-phase `setState` guarded by `disabled !== wasDisabled`. It converges in one extra render before children commit and is StrictMode-safe (the guard prevents any loop). Moving it into `useEffect` is the variant React explicitly recommends against here: the popover would paint once in its open state before the effect runs and closes it.

File: apps/prairielearn/src/components/FileBrowser.tsx [hunk 3]
```diff
@@ -363,79 +173,85 @@ function FileBrowserActions({
...
-          `}
+  return (
+    <div className="d-flex flex-wrap gap-2">
+      {isReadOnly ? null : (
+        <>
+          <a
+            tabIndex={0}
+            className={`btn btn-sm btn-light ${fileInfo.canEdit ? '' : 'disabled'}`}
+            href={`${paths.urlPrefix}/file_edit/${encodedPath}`}
+          >
```
[the remark this answers is not in the document]
Comment (reteps) [reply apps/prairielearn/src/components/FileBrowser.tsx]: Fixed in e1f5d8fdf0. Edit/Download now render through `FileBrowserActionButton`, which emits a real disabled `<button>` when the action is unavailable instead of a CSS-only `.disabled` anchor (the old anchor kept `tabIndex={0}` and stayed keyboard-navigable).

File: apps/prairielearn/src/ee/pages/instructorAiGenerateDraftEditor/components/AiQuestionGenerationEditor.tsx [hunk 4]
```diff
@@ -1,208 +1,266 @@
...
+        staleTime: Infinity,
+        initialData: selectedFilesMatchInitialQuery ? initialQuestionFilesData : undefined,
+        placeholderData: (previousData) => previousData ?? initialQuestionFilesData,
+        retry: 2,
+        retryDelay: (attemptIndex) => Math.min(1000 * 2 ** attemptIndex, 5000),
+      },
+    ),
+  });
+
+  const currentQuestionFilesData = questionFilesData ?? initialQuestionFilesData;
```
[the remark this answers is not in the document]
Comment (reteps) [reply apps/prairielearn/src/ee/pages/instructorAiGenerateDraftEditor/components/AiQuestionGenerationEditor.tsx:130]: Traced this — not a correctness issue, keeping `placeholderData`.

- The editor is remounted on selection change: `AllQuestionFiles` renders `<SelectedQuestionFileEditor key={`${selectedFile.path}:${selectedFile.contentHash}`} />`, so it always resets to whatever file is current.
- The single-file save path is hash-guarded: `SelectedQuestionFileEditor.save()` sends `origHash: selectedFile.contentHash`, and the server rejects a stale save with `STALE_EDIT`.

So during a `placeholderData` transition the displayed file is internally consistent (path + hash + contents all from the same snapshot), and a save targets that file — it can't clobber a different selection. Dropping `placeholderData` would just add flicker on every navigation. (The multi-file `saveDraftQuestionFiles` path is the one without a hash contract — that's the documented TODO in the other thread.)

File: apps/prairielearn/src/ee/pages/instructorAiGenerateDraftEditor/components/useQuestionHtml.ts [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+
+      const form = target;
+      const submitEvent = e as SubmitEvent;
+      const formData = new FormData(form);
+
+      // Copy over the submitter button's name/value if present.
+      const submitter = submitEvent.submitter;
+      if (submitter instanceof HTMLButtonElement && submitter.name && submitter.value) {
+        formData.append(submitter.name, submitter.value);
+      }
```
[the remark this answers is not in the document]
Comment (reteps) [reply apps/prairielearn/src/ee/pages/instructorAiGenerateDraftEditor/components/useQuestionHtml.ts]: Fixed in e1f5d8fdf0 — the check now also accepts `HTMLInputElement` (`<input type="submit">`). Kept the `submitter.value` guard so an input with no explicit value (whose `.value` defaults to a localized "Submit") isn't appended.

File: apps/prairielearn/src/ee/pages/instructorAiGenerateDraftEditor/components/useQuestionHtml.ts [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+  const newVariant = useCallback(() => {
+    refreshPreview({ errorMessage: 'Failed to load a new question variant.' });
+  }, [refreshPreview]);
+
+  const handleNewVariantButtonClick = useCallback(
+    (e: Event) => {
+      if (e.target instanceof HTMLElement && e.target.classList.contains('js-new-variant-button')) {
+        e.preventDefault();
+        newVariant();
+      }
```
[the remark this answers is not in the document]
Comment (reteps) [reply apps/prairielearn/src/ee/pages/instructorAiGenerateDraftEditor/components/useQuestionHtml.ts:268]: Fixed in e1f5d8fdf0 — now matches with `e.target.closest('.js-new-variant-button')` so clicks on nested elements count.

File: apps/prairielearn/src/lib/draft-question-files/mutations.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (reteps) [reply apps/prairielearn/src/lib/draft-question-files/mutations.ts]: Leaving this as the documented `TODO` in the `saveDraftQuestionFiles` JSDoc — threading per-file content hashes is follow-up work beyond this PR's scope. The JSDoc captures the gap, so no separate tracking issue for now.

File: apps/prairielearn/src/tests/e2e/fileEditor.spec.ts [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+}
+
+function questionFilePath(testCoursePath: string, fileName: string) {
+  return path.join(testCoursePath, 'questions', 'addNumbers', fileName);
+}
+
+async function syncCourses(page: Page) {
+  await page.goto('/pl/loadFromDisk');
+  await expect(page).toHaveURL(/\/jobSequence\//);
+  await expect(page.locator('.badge', { hasText: 'Success' })).toBeVisible();
```
[the remark this answers is not in the document]
Comment (reteps) [reply apps/prairielearn/src/tests/e2e/fileEditor.spec.ts]: Fixed in e1f5d8fdf0 — `page.locator('.badge', { hasText: 'Success' })` → `page.getByText('Success', { exact: true })`.

**Comment by reteps** [comment]
@coderabbitai Addressed the 4 nitpicks in e1f5d8fdf0:

- **`QuestionTitleAndQid.tsx` — `aria-errormessage`**: Fixed. The input now uses `aria-errormessage` instead of `aria-describedby` for the inline validation text. (Note: there's only one `<input>` in this file — `InlineEditableField` is reused for both title and QID — so the "also applies to 228-231" range is the error `<div>`, which already has `id={errorId}` + `role="alert"`.)
- **`useDraftQuestionFileMutations.ts` — `useMemo`**: Fixed. Dropped the `useMemo` and return the handler object directly. Referential stability wasn't load-bearing — the consumer doesn't depend on it, and the downstream popover buttons aren't memoized.
- **`aiDraftFileEditor.spec.ts` — class locators**: Partially. The `.badge` selector is fixed (→ `getByText`). Kept `.ace_editor` in `selectedFileEditor()`: it's ACE's own structural DOM hook, not a styling class, and `setAceEditorContentAt` requires a `Locator` pointing at exactly that element to call `ace.edit(el)`. The shared `aceUtils.ts` helpers locate ACE the same way.
- **`editors.ts` — `catch {}`**: Keeping as-is. The error isn't actually dropped — `executeWithServerJob` runs through the server-job framework, which records the failure (status + error message/stack) on the job record before rethrowing. `runEditorJob` returns the `jobSequenceId`, and callers surface it so the failure is viewable on the job sequence page. The bare `catch` is the documented intent ("editor failures are expected ... surfaced via the returned status").

File: apps/prairielearn/assets/scripts/lib/verboseToggle.ts [hunk 8]
```diff
@@ -1,12 +1,15 @@
-import { onDocumentReady } from '@prairielearn/browser-utils';
+// Toggles the visibility of verbose (faint) lines of job output. Uses event
+// delegation on `document` so it also works for job output that is rendered
+// after page load, e.g. by a hydrated React component.
```
Comment (nwalters512) [inline apps/prairielearn/assets/scripts/lib/verboseToggle.ts]: Consider using `delegated-events` instead for consistency, we do that elsewhere. `selector-observer` would also work.

That said: what's the use case for having this interop with React?

File: apps/prairielearn/src/trpc/course/trpc.ts [hunk 9]
```diff
@@ -1,13 +1,15 @@
...
 import { handleTrpcError } from '../../lib/trpc.js';
+import { aiDraftFilesRouter } from '../shared/ai-draft-files.js';
 
 import { courseStaffRouter } from './course-staff.js';
 import { createContext, t } from './init.js';
 import { questionsRouter } from './questions.js';
 import { sharingRouter } from './sharing.js';
 
 const courseRouter = t.router({
+  aiDraftFiles: aiDraftFilesRouter,
```
Comment (nwalters512) [inline apps/prairielearn/src/trpc/course/trpc.ts:14]: I'm wondering if we can get away with this being mounted just here, on the course router? AFAICT the `list` query is the only thing that relies on tRPC itself being context-aware.

I guess there's also authz: the course instance route is impersonation/override-aware, and we'd lose that if we move this behind the course router. But I think we don't actually care about that.

File: apps/prairielearn/src/tests/fileEditor.test.ts [hunk 10]
```diff
@@ -411,12 +405,11 @@ function findEditUrl(name: string, selector: string, url: string, expectedEditUr
...
-function verifyEdit(
-  expectedToFindResults: boolean,
-  expectedToFindChoice: boolean,
-  expectedDraftContents: string,
-  expectedDiskContents: string | null,
-) {
+// Verifies the structure of the editor page. The editor renders its file
+// contents client-side with ACE, which a cheerio test cannot see; on-disk
+// contents are asserted separately with `verifyFileOnLiveDisk`, and the
+/** client-side editor behavior is covered by `tests/e2e/fileEditor.spec.ts`. */
```
Comment (nwalters512) [inline apps/prairielearn/src/tests/fileEditor.test.ts]: Something funky going on with comments here?



```suggestion
// Verifies the structure of the editor page. The editor renders its file
// contents client-side with ACE, which a cheerio test cannot see; on-disk
// contents are asserted separately with `verifyFileOnLiveDisk`, and the
// client-side editor behavior is covered by `tests/e2e/fileEditor.spec.ts`.
```

File: apps/prairielearn/src/pages/jobSequence/jobSequence.html.tsx [hunk 11]
```diff
@@ -40,7 +48,25 @@ export function JobSequence({
...
+          })),
+        },
+        'job-sequence-results-data',
+      )}
+      ${renderHtml(
+        <JobSequenceResults
+          jobSequence={StaffJobSequenceWithJobsSchema.parse(job_sequence)}
+          timeZone={resLocals.course?.display_timezone || 'UTC'}
+        />,
+      )}
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/jobSequence/jobSequence.html.tsx]: Should we consider a more comprehensive refactor here? That is: make `JobSequenceResults` hydratable and manage its own state/rendering in the live-updating case?

We could also save this for a future followup.

File: apps/prairielearn/src/trpc/shared/ai-draft-files.ts [hunk 12]
```diff
@@ -0,0 +1,294 @@
...
+
+export const aiDraftFilesRouter = t.router({
+  list: aiDraftFilesProcedure.input(ListInputSchema).query(async ({ ctx, input }) => {
+    return await getQuestionFilesData({
+      resLocals: { ...ctx.locals, question: ctx.question },
+      editorUrl: `${ctx.locals.urlPrefix}/ai_generate_editor/${ctx.question.id}`,
+      selection: input.selection,
+    });
+  }),
+  save: aiDraftFilesProcedure.input(SaveInputSchema).mutation(async ({ ctx, input }) => {
```
Comment (nwalters512) [inline apps/prairielearn/src/trpc/shared/ai-draft-files.ts]: Why not inline the schemas here? They're only used in this one place, and this makes it harder to see and reason about what a given query/mutation is doing.

File: apps/prairielearn/src/trpc/shared/ai-draft-files.ts [hunk 13]
```diff
@@ -0,0 +1,294 @@
...
+    });
+    if (!input.force && diskHash !== input.origHash) {
+      throwAppError<AiDraftFilesError['Save']>({
+        code: 'STALE_EDIT',
+        message:
+          diskHash == null
+            ? 'This file was deleted since you opened it.'
+            : 'This file changed since you opened it.',
+      });
+    }
```
Comment (nwalters512) [inline apps/prairielearn/src/trpc/shared/ai-draft-files.ts]: I don't love doing this as a separate pre-flight check. While the fact that the file editor machinery checks again under lock means we're ultimately protected against a TOCTOU, it seems like it'd be much better to rely on a a single thing doing that checking, and turning a failure into the right kind of error here.

File: apps/prairielearn/src/trpc/shared/ai-draft-files.ts [hunk 14]
```diff
@@ -0,0 +1,294 @@
...
+}
+
+/**
+ * Typed errors for the draft-file operations the client invokes, keyed by
+ * operation so callers narrow with `getAppError<AiDraftFilesError['<Op>']>`.
+ * These are the `aiDraftFiles` tRPC procedures plus `Upload`, which is served by
+ * a multipart route rather than tRPC (tRPC has no multipart support) but shares
+ * the same `SYNC_JOB_FAILED` error contract. An operation with no typed errors
+ * is declared `never` (resolving client-side to just the `UNKNOWN` branch), so
+ * the map stays a complete picture of the surface.
```
Comment (nwalters512) [inline apps/prairielearn/src/trpc/shared/ai-draft-files.ts]: I'd been looking at https://trpc.io/docs/server/non-json-content-types - can we leverage that here? This came up with the QTI import stuff as well. It'd be great if we could funnel everything through tRPC.

File: apps/prairielearn/src/tests/e2e/utils/ace.ts
Comment (nwalters512) [inline apps/prairielearn/src/tests/e2e/utils/ace.ts:1]: Aside / note to self: I think I want a separate `utils/` directory in `e2e`. Might add that on master and then update this branch.

File: apps/prairielearn/src/pages/instructorFileEditor/instructorFileEditor.tsx [hunk 15]
```diff
@@ -179,11 +177,133 @@ router.get(
...
+// `readDraftEdit`/`writeDraftEdit`/`updateJobSequenceId` helpers below exist
+// only to carry the editor's contents and the sync job's result across that
+// redirect — `instructorFileEditor` is the sole consumer of `file_edits`.
+//
+// Converting this save to a tRPC mutation that stays on the
+// page would let the client keep its contents in React state and receive the
+// job result as a return value, retiring the `file_edits` mechanism, the
+// `DraftEdit` type, and the `versionChoice`/`alertChoice` UI. The concurrent-
+// edit case is already handled cleanly by the `STALE_EDIT` flow in
+// `trpc/shared/ai-draft-files.ts`.
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorFileEditor/instructorFileEditor.tsx]: This seems like a sensible direction. Can you open a tracking issue for this, and maybe parent it under the AI editing issue? It's not exactly the same stream of work but I think it's the best home to track followups like this.

File: apps/prairielearn/src/tests/e2e/aiDraftFileEditor.spec.ts [hunk 16]
```diff
@@ -0,0 +1,166 @@
...
+test.describe.configure({ mode: 'serial', timeout: 60_000 });
+
+test.describe('AI draft file editor', () => {
+  let editorUrl: string;
+  let questionQid: string;
+
+  test.beforeAll(async ({ browser, workerPort }) => {
+    const page = await browser.newPage({ baseURL: `http://localhost:${workerPort}` });
+    await syncCourses(page);
+    await page.close();
```
Comment (nwalters512) [inline apps/prairielearn/src/tests/e2e/aiDraftFileEditor.spec.ts]: IMO this setup is not correct - this seems to be mutating the test course directly? We should be using something like the `testCoursePath` fixture right? Ditto for the other test here.

That said I'm also not convinced that our existing fixtures are exactly correct yet, I think they're worker scoped and require manual cleanup? This might be something we want to consider and improve on master first.

File: apps/prairielearn/src/pages/instructorFileEditor/instructorFileEditor.tsx [hunk 17]
```diff
@@ -179,11 +177,133 @@ router.get(
...
+                    ><code>${unsafeHtml(
+                      ansiToHtml(editorData.fileMetadata.syncWarnings),
+                    )}</code></pre>
+                  </div>
+                `
+              : ''}
+            <h1 class="visually-hidden">File editor</h1>
+
+            ${renderHtml(
+              <Hydrate>
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/instructorFileEditor/instructorFileEditor.tsx]: Consider using `hydrateHtml` here.

**Review by nwalters512 — COMMENTED** [review]
This is not yet a comprehensive review, but I wanted to still pass along my feedback so far to give you some things to consider.

File: apps/prairielearn/src/trpc/course/trpc.ts (hunk 9 above)
Comment (reteps) [reply apps/prairielearn/src/trpc/course/trpc.ts:14]: Sounds good, we will do this.

File: apps/prairielearn/src/pages/jobSequence/jobSequence.html.tsx (hunk 11 above)
Comment (reteps) [reply apps/prairielearn/src/pages/jobSequence/jobSequence.html.tsx]: Followup.

File: apps/prairielearn/assets/scripts/lib/verboseToggle.ts (hunk 8 above)
Comment (reteps) [reply apps/prairielearn/assets/scripts/lib/verboseToggle.ts]: This is for the job sequence results page.

File: apps/prairielearn/src/trpc/shared/ai-draft-files.ts (hunk 13 above)
Comment (reteps) [reply apps/prairielearn/src/trpc/shared/ai-draft-files.ts]: self-TODO: investigate this

File: apps/prairielearn/src/trpc/shared/ai-draft-files.ts (hunk 14 above)
Comment (reteps) [reply apps/prairielearn/src/trpc/shared/ai-draft-files.ts]: self-TODO: discuss if this is a blocker

File: apps/prairielearn/assets/scripts/lib/verboseToggle.ts (hunk 8 above)
Comment (reteps) [reply apps/prairielearn/assets/scripts/lib/verboseToggle.ts]: fixed in f100a4946

File: apps/prairielearn/src/trpc/course/trpc.ts (hunk 9 above)
Comment (reteps) [reply apps/prairielearn/src/trpc/course/trpc.ts:14]: fixed in f100a4946

File: apps/prairielearn/src/tests/fileEditor.test.ts (hunk 10 above)
Comment (reteps) [reply apps/prairielearn/src/tests/fileEditor.test.ts]: fixed in f100a4946

File: apps/prairielearn/src/pages/jobSequence/jobSequence.html.tsx (hunk 11 above)
Comment (reteps) [reply apps/prairielearn/src/pages/jobSequence/jobSequence.html.tsx]: https://github.com/PrairieLearn/PrairieLearn/issues/15129

File: apps/prairielearn/src/trpc/shared/ai-draft-files.ts (hunk 12 above)
Comment (reteps) [reply apps/prairielearn/src/trpc/shared/ai-draft-files.ts]: fixed.

File: apps/prairielearn/src/trpc/shared/ai-draft-files.ts (hunk 13 above)
Comment (reteps) [reply apps/prairielearn/src/trpc/shared/ai-draft-files.ts]: agreed, this follows how the existing editor works. fixed.

File: apps/prairielearn/src/trpc/shared/ai-draft-files.ts (hunk 14 above)
Comment (reteps) [reply apps/prairielearn/src/trpc/shared/ai-draft-files.ts]: #15130

File: apps/prairielearn/src/tests/e2e/utils/ace.ts
Comment (reteps) [reply apps/prairielearn/src/tests/e2e/utils/ace.ts:1]: done in d8aae31e5a

File: apps/prairielearn/src/pages/instructorFileEditor/instructorFileEditor.tsx (hunk 15 above)
Comment (reteps) [reply apps/prairielearn/src/pages/instructorFileEditor/instructorFileEditor.tsx]: https://github.com/PrairieLearn/PrairieLearn/issues/15100

File: apps/prairielearn/src/tests/e2e/aiDraftFileEditor.spec.ts (hunk 16 above)
Comment (reteps) [reply apps/prairielearn/src/tests/e2e/aiDraftFileEditor.spec.ts]: fixed in f100a4946

File: apps/prairielearn/src/pages/instructorFileEditor/instructorFileEditor.tsx (hunk 17 above)
Comment (reteps) [reply apps/prairielearn/src/pages/instructorFileEditor/instructorFileEditor.tsx]: fixed in f100a4946
